### ISSUE TITLE
False positive on Layout/EndAlignment (when `EnforcedStyleAlignWith` is set to variable)

### ISSUE THREAD

**Issue by Drowze** [rubocop/rubocop#14667]
_NOTE: this is possibly the related to #12297_

## Steps to reproduce the problem

```yaml
# .rubocop.yml
Layout/EndAlignment:
  EnforcedStyleAlignWith: variable
```

```ruby
# main.rb
# does not trigger Layout/EndAlignment cop (as expected?)
value = if rand(0..1).zero?
  [1, 2, 3].first
end.to_i
puts value

# triggers Layout/EndAlignment cop (bug?)
value = if rand(0..1).zero?
  [1, 2, 3].first
end&.to_i
puts value
```

```
$ rubocop --only Layout/EndAlignment main.rb
```

## Expected behavior

No offenses reported.

## Actual behavior

```
Offenses:

main.rb:9:3: C: [Correctable] Layout/IndentationWidth: Use 2 (not -6) spaces for indentation.
  [1, 2, 3].first
  ^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

## RuboCop version

```
$ rubocop -V
1.81.7 (using Parser 3.3.10.0, rubocop-ast 1.48.0, analyzing as Ruby 2.7, running on ruby 3.4.7) [arm64-darwin24]
```
